src/components/DailyScheduler.js

```diff
@@ -0,0 +1,224 @@
+import React, { useState } from "react";
+import { useSelector, useDispatch } from "react-redux";
+import DatePicker from "react-datepicker";
+import { add, isSameDay, isEqual, parseISO } from "date-fns";
+import ko from "date-fns/locale/ko";
+import styled from "styled-components";
+
+import DateSchedule from "./DateSchedule";
+import Modal from "./common/Modal";
+import { updateAvailableDates } from "../features/counselorSlice";
+import {
+  TYPE_DATE,
+  EXISTED_TIMELINE,
+  START_DATE,
+  END_DATE,
+  DATE_SELECTION,
+  TIME_SELECTION,
+} from "../constants/date";
+
+import "../assets/stylesheets/timepicker-custom.css";
+
+function DailyScheduler() {
+  const dispatch = useDispatch();
+  const userId = useSelector(({ user }) => user.data._id);
+  const scheduleList = useSelector(
+    ({ counselor }) => counselor.data.availableDates
+  ).filter(({ type }) => type === TYPE_DATE);
+
+  const [selectedDate, setSelectedDate] = useState("");
+  const [selectedTime, setSelectedTime] = useState({});
+  const [availableDates, setAvailableDates] = useState({ type: TYPE_DATE });
+  const [errorMessage, setErrorMessage] = useState(null);
+
+  function handleClickDate(date) {
+    setSelectedDate(date);
+    setSelectedTime({
+      startDate: "",
+      endDate: "",
+    });
+  }
+
+  function handleClickSaveButton() {
+    if (!selectedDate || !availableDates.startDate || !availableDates.endDate) {
+      setErrorMessage(TIME_SELECTION);
+      return;
+    }
+
+    dispatch(
+      updateAvailableDates({
+        userId,
+        availableDates,
+      })
+    );
+
+    setAvailableDates({ type: TYPE_DATE });
+  }
+
+  function handleStartTimeToUtc(time) {
+    const addedStartTime = add(selectedDate, {
+      hours: time.getHours(),
+    });
+
+    setSelectedTime({
+      startDate: addedStartTime,
+    });
+
+    setAvailableDates((date) => ({
+      ...date,
+      startDate: addedStartTime,
+    }));
+  }
+
+  function handleEndTimeToUtc(time) {
+    const addedEndTime = add(selectedDate, {
+      hours: time.getHours(),
+    });
+
+    handleClickEndTime(time);
+
+    setSelectedTime((time) => ({
+      ...time,
+      endDate: addedEndTime,
+    }));
+
+    setAvailableDates((date) => ({
+      ...date,
+      endDate: addedEndTime,
+    }));
+  }
```

Comment:
```suggestion
  function handleOnStartTimeChange(time) {
    const addedStartTime = add(selectedDate, {
      hours: time.getHours(),
    });

    setSelectedTime({
      startDate: addedStartTime,
    });

    setAvailableDates((date) => ({
      ...date,
      startDate: addedStartTime,
    }));
  }

  function handleOnEndTimeChange(time) {
    const addedEndTime = add(selectedDate, {
      hours: time.getHours(),
    });

    handleClickEndTime(time);

    setSelectedTime((time) => ({
      ...time,
      endDate: addedEndTime,
    }));

    setAvailableDates((date) => ({
      ...date,
      endDate: addedEndTime,
    }));
  }
```

Datepicker를 핸들링하는 일반적인 네이밍으로 들어가야 할 것 같아요! 
안에서 이루어지는 로직들이 단지 UTC만으로 변경하는 것이 아니라 다른 로직들도 포함되어 있어서요!